tests/php/Unit/WC/Gateway/Mollie_WC_Gateway_AbstractTest.php

```diff
@@ -214,5 +214,193 @@ public function associativePaymentMethodsImagesReturnsEmptyArrayIfApiFails()
+        list(
+            $orderId, $orderKey, $homeUrl, $apiRequestUrl, $untrailedUrl,
+            $urlWithParams,
+            $untrailedWithParams
+            )
+            = $this->testPolylangVariables();
+
+        /*
+        * Setup Stubs
+        */
+        $wcUrl = $this->createConfiguredMock(
+            \WooCommerce::class,
+            ['api_request_url' => $apiRequestUrl]
+        );
+        $wcOrder = $this->createMock('WC_Order');
+
+
+        /*
+        * Expectations
+        */
+        expect('get_home_url')
+            ->andReturn($homeUrl);
+        //get url from request
+        expect('WC')
+            ->andReturn($wcUrl);
+        //delete url final slash
+        expect('untrailingslashit')
+            ->twice()
+            ->andReturn($untrailedUrl, $untrailedWithParams);
+        //get order id and key and append to the the url
+        expect('wooCommerceOrderId')
+            ->andReturn($orderId);
+        expect('wooCommerceOrderKey')
+            ->andReturn($orderKey);
+        $testee
+            ->expects($this->once())
+            ->method('appendOrderArgumentsToUrl')
+            ->with($orderId, $orderKey, $untrailedUrl)
+            ->willReturn($urlWithParams);
+        //check for multilanguage plugin enabled, receives url and adds it
+        $testee
+            ->expects($this->once())
+            ->method('getSiteUrlWithLanguage')
+            ->willReturn("{$homeUrl}/nl");
+        expect('debug')
+            ->withAnyArgs();
+
+        /*
+         * Execute test
+         */
+        $result = $testee->getWebhookUrl($wcOrder);
+
+        self::assertEquals(
+            "{$homeUrl}/nl/wc-api/mollie_return/?order_id={$orderId}&key=wc_order_{$orderKey}",
+            $result
+        );
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function testPolylangTestee()
+    {
+        $testee = $this
+            ->buildTesteeMock(
+                Testee::class,
+                [],
+                ['getSiteUrlWithLanguage', 'appendOrderArgumentsToUrl']
+            )
+            ->getMockForAbstractClass();
+        $testee = $this->proxyFor($testee);
+        return $testee;
+    }
 
+    /**
+     * @return array
+     */
+    protected function testPolylangVariables()
+    {
+        $orderId = $this->faker->randomDigit;
+        $orderKey = $this->faker->word;
+        $homeUrl = rtrim($this->faker->url, '/\\');
+        $apiRequestUrl = "{$homeUrl}/wc-api/mollie_return";
+        $untrailedUrl = rtrim($apiRequestUrl, '/\\');
+        $urlWithParams
+            = "{$untrailedUrl}/?order_id={$orderId}&key=wc_order_{$orderKey}";
+        $untrailedWithParams = rtrim($urlWithParams, '/\\');
+        return array(
+            $orderId,
+            $orderKey,
+            $homeUrl,
+            $apiRequestUrl,
+            $untrailedUrl,
+            $urlWithParams,
+            $untrailedWithParams
+        );
+    }
```

Comment:
If these are always the same and you are not interested in producing new values would be more convenient to have them as properties of the class. You can avoid the cost of producing the array for then unpack it in the test.

I usually do:

```
protected function setUp()
{
     parent::setUp();

     $this->initializeStubs();
}

protected function initializeStubs()
{
        $this->orderId = $this->faker->randomDigit;
        $this->orderKey = $this->faker->word;
        $this->homeUrl = rtrim($this->faker->url, '/\\');
        $this->apiRequestUrl = "{$homeUrl}/wc-api/mollie_return";
        $this->untrailedUrl = rtrim($apiRequestUrl, '/\\');
        $this->urlWithParams = "{$untrailedUrl}/?order_id={$orderId}&key=wc_order_{$orderKey}";
        $this->untrailedWithParams = rtrim($urlWithParams, '/\\');
}
```